dask/bytes/tests/test_s3.py

```diff
@@ -193,32 +201,32 @@ def test_getsize(fmt):
 
 def test_modification_time_read_bytes():
     with s3_context('compress', files) as s3:
-        _, a = read_bytes('compress/test/accounts.*', s3=s3)
-        _, b = read_bytes('compress/test/accounts.*', s3=s3)
+        _, a = read_bytes('compress/test/accounts.*', storage_options={'s3': s3})
+        _, b = read_bytes('compress/test/accounts.*', storage_options={'s3': s3})
```

Comment:
@mrocklin I would like to start with that it seems to be your idea of bringing `storage_options` ([commit](https://github.com/dask/dask/commit/3ccf281168620d4075f89e7c6f8c03f45c0e810b)). Are you in conflict with yourself? :) My patch just extends the idea on all IO APIs, so they are "aligned".

I have been contemplating improvements to this API since I bumped into the issue related to this (#1232), yet I haven't found a better way.